src/execution/const_interpreter_loop.rs

```diff
@@ -0,0 +1,101 @@
+
+        match first_instr_byte {
+            END => {
+                break;
+            }
+            I32_CONST => {
+                let constant = wasm.read_var_i32().unwrap_validated();
+                trace!("Constant instruction: i32.const [] -> [{constant}]");
+                stack.push_value(constant.into());
+            }
+            I32_ADD => {
+                let v1: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                let v2: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                let res = v1.wrapping_add(v2);
+
+                trace!("Constant instruction: i32.add [{v1} {v2}] -> [{res}]");
+                stack.push_value(res.into());
+            }
+            I32_SUB => {
+                let v2: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                let v1: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                let res = v1.wrapping_sub(v2);
+
+                trace!("Constant instruction: i32.sub [{v1} {v2}] -> [{res}]");
+                stack.push_value(res.into());
+            }
+            I32_MUL => {
+                let v1: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                let v2: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                let res = v1.wrapping_mul(v2);
+
+                trace!("Constant instruction: i32.mul [{v1} {v2}] -> [{res}]");
+                stack.push_value(res.into());
+            }
+            I64_CONST => {
+                let constant = wasm.read_var_i64().unwrap_validated();
+                trace!("Constant instruction: i64.const [] -> [{constant}]");
+                stack.push_value(constant.into());
+            }
+            I64_ADD => {
+                let v1: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                let v2: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                let res = v1.wrapping_add(v2);
+
+                trace!("Constant instruction: i64.add [{v1} {v2}] -> [{res}]");
+                stack.push_value(res.into());
+            }
+            I64_SUB => {
+                let v2: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                let v1: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                let res = v1.wrapping_sub(v2);
+
+                trace!("Constant instruction: i64.sub [{v1} {v2}] -> [{res}]");
+                stack.push_value(res.into());
+            }
+            I64_MUL => {
+                let v1: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                let v2: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                let res = v1.wrapping_mul(v2);
+
+                trace!("Constant instruction: i64.mul [{v1} {v2}] -> [{res}]");
+                stack.push_value(res.into());
+            }
+            _ => panic!("¯\\_(ツ)_/¯"),

```

Comment:
Please add a proper "invalid instruction" message. You can keep the lil guy here

Reply:
In the main interpreter loop, we put out a trace message and just skip the instruction. Should we do that here too, instead of panic?

Reply:
Hmm.. I think that panicking both here and in the interpreter_loop might be better to catch errors ASAP.
Please leave this as a panic for now. I will create an issue for the interpreter_loop.

Reply:
Okay, modified it in both places. What do you think?
In regards the the little buddy, I've talked to him and convinced him to free up the spot. Knowing him, he will come back, but not in this PR.